cookbooks/bcpc-hadoop/recipes/resource_manager.rb

```diff
@@ -32,22 +32,54 @@
   not_if "hdfs dfs -test -d /var/log/hadoop-yarn", :user => "hdfs"
 end
 
+bash "create-hdfs-yarn-ats-log" do
+  code <<-EOH
+  hdfs dfs -mkdir -p /var/log/ats && hdfs dfs -chmod 1777 /var/log/ats && hdfs dfs -chown yarn:mapred /var/log/ats
+  hdfs dfs -mkdir -p /var/log/ats/active && hdfs dfs -chmod 1777 /var/log/ats/active && hdfs dfs -chown yarn:mapred /var/log/ats/active
+  hdfs dfs -mkdir -p /var/log/ats/done && hdfs dfs -chmod 0700 /var/log/ats/done && hdfs dfs -chown yarn:mapred /var/log/ats/done
+  EOH
+  user "hdfs"
+  only_if { node.roles.include?("BCPC-Hadoop-Head-YarnTimeLineServer") }
```

Comment:
Just a question: Does searching for the role seem "better" than looking in the recipe collection? (A nice [discussion](https://gist.github.com/kwilczynski/c271f251ea7f48ba99b3) of the various API's and semantics.)